jni_gen/lib/src/bindings/dart_bindings.dart

```diff
@@ -0,0 +1,380 @@
+  }
+
+  String _field(ClassDecl c, Field f) {
+    final name = f.finalName;
+    final s = StringBuffer();
+
+    void _writeDocs({bool writeDeleteInstruction = true}) {
+      s.write('$_indent/// from: ${_originalFieldDecl(f)}\n');
+      if (!isPrimitive(f.type) && writeDeleteInstruction) {
+        s.write(_deleteInstruction);
+      }
+      s.write(_breakDocComment(f.javadoc));
+    }
+
+    if (isStaticField(f) && isFinalField(f) && f.defaultValue != null) {
+      _writeDocs(writeDeleteInstruction: false);
+      s.write('${_indent}static const $name = ${_literal(f.defaultValue)};\n');
+      return s.toString();
+    }
+    final cName = memberNameInC(c, name);
+
+    void writeAccessor({bool isSetter = false}) {
+      final symPrefix = isSetter ? 'set' : 'get';
+      final sym = '_$symPrefix$name';
+      final ffiSig = dartSigForField(f, isSetter: isSetter, isFfiSig: true);
+      final dartSig = dartSigForField(f, isSetter: isSetter, isFfiSig: false);
+      s.write('${_indent}static final $sym = $_jlookup'
+          '<${ffi}NativeFunction<$ffiSig>>("${symPrefix}_$cName")\n'
+          '.asFunction<$dartSig>();\n');
+      // write original type
+      _writeDocs();
+      s.write(_indent);
+      if (isStaticField(f)) s.write('static ');
+      if (isSetter) {
+        s.write('set $name(${dartOuterType(f.type)} value) => $sym(');
+        if (!isStaticField(f)) {
+          s.write('$_self, ');
+        }
+        s.write(_toCArg('value', f.type));
+        s.write(');\n');
+      } else {
+        // getter
+        final self = isStaticField(f) ? '' : _self;
+        final outer = dartOuterType(f.type);
+        final callExpr = '$sym($self)';
+        final resultExpr = _toDartResult(callExpr, f.type, outer);
+        s.write('$outer get $name => $resultExpr;\n');
+      }
+    }
+
+    writeAccessor(isSetter: false);
+    if (!isFinalField(f)) writeAccessor(isSetter: true);
+    return s.toString();
+  }
+
+  String _getSimpleName(String binaryName) {
+    final components = binaryName.split(".");
+    return components.last.replaceAll("\$", "_");
+  }
+
+  String dartSigForField(Field f,
+      {bool isSetter = false, required bool isFfiSig}) {
+    final conv = isFfiSig ? dartFfiType : dartInnerType;
+    final voidType = isFfiSig ? _void : 'void';
+    final ref = f.modifiers.contains('static') ? '' : '$_voidPtr, ';
+    if (isSetter) {
+      return '$voidType Function($ref${conv(f.type)})';
+    }
+    return '${conv(f.type)} Function($ref)';
+  }
+
+  String dartSigForMethod(Method m, {required bool isFfiSig}) {
+    final conv = isFfiSig ? dartFfiType : dartInnerType;
+    final argTypes = [if (hasSelfParam(m)) _voidPtr];
+    for (var param in m.params) {
+      argTypes.add(conv(param.type));
+    }
+    final retType = isCtor(m) ? _voidPtr : conv(m.returnType);
+    return '$retType Function (${argTypes.join(", ")})';
+  }
+
+  // Type for FFI Function signature
+  String dartFfiType(TypeUsage t) {
```

Comment:
I think it can remain along with other methods in binding generator class, because 1) It's not used elsewhere, 2) Future changes may make it require some more context eg: from options.